tools/cloud-build/daily-tests/remove_module.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import yaml # pip install pyyaml
+
+def delete_module(bp, mod_id: str):
+  for g in bp["deployment_groups"]:
+    for im, m in enumerate(g["modules"]):
+      if m["id"] == mod_id:
+        del g["modules"][im]
+        return
+  raise RuntimeError(f"Module {mod_id} not found")
+
+def update_bp(bp, mod_id: str):
+  delete_module(bp, mod_id)
+   # remove module from "use"
+  for g in bp["deployment_groups"]:
+    for m in g["modules"]:
+      if mod_id in m.get("use", []):
+        m["use"].remove(mod_id)
+  # TODO:
+  # - move all settings of removed module into some "/dev/null" to prevent
+  #   "variable not used" validation errors
+  # - treat references to removed module (how ?)
```

Comment:
IIUC there is no handling of `$(<module_id>.<output>)` or the like. 

1. Add description and usage instructions. Include written warning in description that includes this gap in functionality.  
2. Could we do a quick RE search of the BP text to just confirm that there are no explicit references and error if there are. Could even look for `<module_id>.`. Maybe this is not needed because it will be caught by ghpc in following steps, but it would be clearer if the error came from the module removal step.